api/src/main/java/org/openmrs/module/appointments/advice/RecurringAppointmentsAdvice.java

```diff
@@ -0,0 +1,79 @@
+package org.openmrs.module.appointments.advice;
+
+import org.openmrs.api.context.Context;
+import org.openmrs.module.appointments.model.Appointment;
+import org.openmrs.module.appointments.model.AppointmentRecurringPattern;
+
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.openmrs.module.appointments.constants.AppointmentsEventRecordsConstants.CATEGORY;
+import static org.openmrs.module.appointments.constants.AppointmentsEventRecordsConstants.RAISE_EVENT_GLOBAL_PROPERTY;
+
+public class RecurringAppointmentsAdvice extends AbstractBaseAdvice {
+
+    private static final String TITLE = "RecurringAppointments";
+    private static final String VALIDATE_AND_SAVE = "validateAndSave";
+    private static final String UPDATE = "update";
+    private static final String CHANGE_STATUS = "changeStatus";
+    private static final List<String> METHOD_NAMES = Arrays.asList(VALIDATE_AND_SAVE, UPDATE, CHANGE_STATUS);
+    private static final String URL_PATTERN_GLOBAL_PROPERTY = "atomfeed.event.urlPatternForRecurringAppointments";
+    private static final String DEFAULT_URL_PATTERN = "/openmrs/ws/rest/v1/recurring-appointments?uuid={uuid}";
+
+    @Override
+    public void afterReturning(Object returnValue, Method method, Object[] arguments, Object target) throws Throwable {
+        if (VALIDATE_AND_SAVE.equals(method.getName()) || isAllRecurringAppointmentsUpdate(method, returnValue)) {
+            AppointmentRecurringPattern appointmentRecurringPattern = (AppointmentRecurringPattern) returnValue;
+            raiseEventsForAppointments(method, arguments, target, new ArrayList<>(appointmentRecurringPattern.getAppointments()));
```

Comment:
raiseEventsForAppointments always is called, only what list of appointments we pass changes. Can we bring it out commonly?

Reply:
Done !